public/components/layer_config/documents_config/document_layer_style.tsx

```diff
@@ -37,29 +61,38 @@ export const DocumentLayerStyle = ({ setSelectedLayerConfig, selectedLayerConfig
   );
 
   useEffect(() => {
-    setFillColor(selectedLayerConfig?.style?.fillColor);
-    setBorderColor(selectedLayerConfig?.style?.borderColor);
+    setFillColor(
+      selectedLayerConfig?.style?.fillColor,
+      !!fillColorErrors ? { isValid: false } : { isValid: true }
+    );
+    setBorderColor(
+      selectedLayerConfig?.style?.borderColor,
+      !!borderColorErrors ? { isValid: false } : { isValid: true }
+    );
   }, [selectedLayerConfig]);
```

Comment:
why not change if borderColor & fillColor is updated?

Reply:
This useEffect is used to update the style color when switch layer config(layer id changes) between different document layers. Please check updated code:

https://github.com/opensearch-project/dashboards-maps/blob/6671f3de46716a0c558db90eb463d0576056dc80/public/components/layer_config/documents_config/document_layer_style.tsx#L63-L73